lfs/transfer_queue.go

```diff
@@ -247,3 +276,62 @@ func (q *TransferQueue) Process() {
 func (q *TransferQueue) Errors() []*WrappedError {
 	return q.errors
 }
+
+// progressLogger provides a wrapper around an os.File that can either
+// write to the file or ignore all writes completely.
+type progressLogger struct {
+	writeData bool
+	log       *os.File
+}
+
+// Write will write to the file and perform a Sync() if writing succeeds.
+func (l *progressLogger) Write(b []byte) error {
+	if l.writeData {
+		if _, err := l.log.Write(b); err != nil {
+			return err
+		}
+		return l.log.Sync()
+	}
+	return nil
+}
+
+// Close will call Close() on the underlying file
+func (l *progressLogger) Close() error {
+	if l.writeData {
+		return l.log.Close()
+	}
```

Comment:
This should check if `l.log` is nil instead so `Close()` actually closes the handle after an error.